init.lua

```diff
@@ -247,10 +247,10 @@ local function smooth(pos1, pos2, deadzone)
 			elseif old_height < new_height then
 				-- need to add nodes
 				local y = old_height
-				local c_target = minetest.get_node(vector.new(index_x, old_height, index_z))
+				local old_height_index = index_z + (offset.y + old_height) * stride.y
```

Comment:
it would be better to fetch `data[old_height_index]` just once

also, there's an edge case: what if old_height is 0?

Reply:
Ah, good points. I've made that performance improvement. Not sure I see that edge-case though - although I'm rather new to Lua (and I've heard it's weird about starting from 1 in some cases).

Reply:
Ahhhhhhh. In that case we need to default to something if there aren't any nodes in a column?